### PR TITLE
Back from the bowels.

### PR DESCRIPTION
**Pull request by Zemurin**
- Now I finally, and for the first time, comprehend what the converter *does*.

### CODE REVIEW

File: EU4ToVic2Tests/MapperTests/CountryMappings/CountryMappingsTests.cpp [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+
+	const Configuration testConfiguration;
+	std::stringstream colonialInput;
+	std::stringstream provinceInput;
+	provinceInput << "0.0.0.0 = {}";
+	mappers::ProvinceMapper provinceMapper(provinceInput, colonialInput, testConfiguration);
+
+	mapper.createMappings(cultureGroupsMapper, countries.getTheCountries(), provinceMapper);
+
+	// Trivial, right?
```
Comment (IhateTrains) [inline EU4ToVic2Tests/MapperTests/CountryMappings/CountryMappingsTests.cpp:35]: 😄

File: EU4toV2/Source/Mappers/CultureGroups/CultureGroups.cpp [hunk 2]
```diff
@@ -44,55 +44,57 @@ mappers::CultureGroups::CultureGroups(std::istream& theStream)
...
-				{
-					cultureGroupsMap[cultureGroupName].mergeCulture(cultureItr.first, cultureItr.second);
-				}
+				cultureGroupsMap[cultureGroupName].mergeCulture(cultureItr.first, cultureItr.second);
 			}
-			else cultureGroupsMap.insert(std::make_pair(cultureGroupName, newGroup));
-		});
+		}
+		else
+			cultureGroupsMap.insert(std::make_pair(cultureGroupName, newGroup));
```
Comment (IhateTrains) [inline EU4toV2/Source/Mappers/CultureGroups/CultureGroups.cpp:60]: `emplace` would work here, AFAIK it's more efficient.

**Review by IhateTrains — APPROVED** [review]

File: EU4toV2/Source/Mappers/CultureGroups/CultureGroups.cpp (hunk 2 above)
Comment (Zemurin) [reply EU4toV2/Source/Mappers/CultureGroups/CultureGroups.cpp:60]: I didn't even edit this file, it's just reclang speaking. I'll get to it in the next pass.

**Review by Idhrendur — COMMENTED** [review]
Partial review, as I'm not feeling well and can't make heads or tails of the more complex code right now.

**Comment by Zemurin** [comment]
> 
> 
> Partial review, as I'm not feeling well and can't make heads or tails of the more complex code right now.

Review fully when you feel better, I have no qualms piling up the commits. I've tried to explain in tests what's going on and more importantly, why, when necessary in the complicated ones.

**Comment by Zemurin** [comment]
c-d section of the mappers has been concluded. 
Of note (outside regular cleanup):
- countrymappings, the stuff that maps eu4 tags to vic2 tags has been drastically reorganized and tested
- culturegroups, an obscure mapper with seemingly trivial function, has been demonstrated in action
- culturemapper has received a full barrage of tests, dumping out that commented block entirely, mocks along with it.
- mods NO LONGER carry name key in .mod file, so we're now hunting for mod name in the descriptor. Someone should tell fronter about this, though neither vic2 nor ck2 fortunately use new launcher.

**Comment by Zemurin** [comment]
e-j section completed. This PR is getting heavy, I warmly recommend forgetting about git diff (it's getting unusable) and just checking it out locally.

File: EU4ToVic2Tests/MapperTests/CultureGroups/CultureTests.cpp [hunk 3]
```diff
@@ -0,0 +1,172 @@
+#include "CultureGroups/Culture.h"
+#include "gtest/gtest.h"
+
```
Comment (Idhrendur) [inline EU4ToVic2Tests/MapperTests/CultureGroups/CultureTests.cpp:3]: Add tests (or a test) that demonstrate(s) the default values for the different variables.

File: EU4toV2/Source/Mappers/CulturalUnions/CulturalUnionMapper.h [hunk 4]
```diff
@@ -1,24 +1,21 @@
...
-		
-		void loadFile(const std::string& filename);
+class CulturalUnionMapper: commonItems::parser
+{
+  public:
+	CulturalUnionMapper(const std::string& filename);
+	explicit CulturalUnionMapper(std::istream& theStream);
 
-		[[nodiscard]] std::optional<std::vector<std::string>> getCoresForCulture(const std::string& culture) const;
+	[[nodiscard]] std::optional<std::vector<std::string>> getCoresForCulture(const std::string& culture) const;
```
Comment (Idhrendur) [inline EU4toV2/Source/Mappers/CulturalUnions/CulturalUnionMapper.h:13]: Optional is unnecessary here. An empty vector should indicate no result.

File: EU4ToVic2Tests/MapperTests/CultureGroups/CultureGroupTests.cpp [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+	input << "someculture = {}\n";
+	input << "otherculture = {}\n";
+	const mappers::CultureGroup group("name", input);
+
+	ASSERT_EQ(2, group.getCultures().size());
+	ASSERT_TRUE(group.getCultures().contains("someculture"));
+	ASSERT_TRUE(group.getCultures().contains("otherculture"));
+}
+
+TEST(Mappers_CultureGroupTests, culturesCanBePinged)
```
Comment (Idhrendur) [inline EU4ToVic2Tests/MapperTests/CultureGroups/CultureGroupTests.cpp:25]: Test the negative case too.

File: EU4toV2/Source/Mappers/CultureGroups/CultureGroup.h [hunk 6]
```diff
@@ -1,39 +1,40 @@
...
-		std::string isOverseas;
-		std::string culturalUnionTag;
-		std::vector<std::string> maleNames;
-		std::vector<std::string> femaleNames;
-		std::vector<std::string> dynastyNames;
-		std::map<std::string, Culture> cultures;
-	};
-}
+	std::string name;
+	std::string graphicalCulture;
```
Comment (Idhrendur) [inline EU4toV2/Source/Mappers/CultureGroups/CultureGroup.h:28]: This (and presumably other variables) doesn't show it the tests. If it matters, make tests show how it does. If it doesn't remove it.

File: EU4toV2/Source/Mappers/FactoryInventions/FactoryInventionMapper.h [hunk 7]
```diff
@@ -1,25 +1,23 @@
...
-		explicit FactoryInventionMapper(std::istream& theStream);
-		
-		[[nodiscard]] const auto& getFactoryInventionMap() const { return factoryInventionMap; }
+class FactoryInventionMapper: commonItems::parser
+{
+  public:
+	FactoryInventionMapper();
+	explicit FactoryInventionMapper(std::istream& theStream);
+
+	[[nodiscard]] const auto& getFactoryInventionMap() const { return factoryInventionMap; }
```
Comment (Idhrendur) [inline EU4toV2/Source/Mappers/FactoryInventions/FactoryInventionMapper.h:14]: Investigate whether it makes sense to return the map, or provide a lookup function instead.

File: EU4toV2/Source/Mappers/FactoryTypes/FactoryType.h [hunk 8]
```diff
@@ -1,41 +1,38 @@
 #ifndef FACTORY_TYPE_H
```
Comment (Idhrendur) [inline EU4toV2/Source/Mappers/FactoryTypes/FactoryType.h:1]: Add tests

File: EU4toV2/Source/Mappers/Geography/ClimateMapper.h [hunk 9]
```diff
@@ -1,28 +1,24 @@
...
-		explicit ClimateMapper(std::istream& theStream);
-		
-		[[nodiscard]] const auto& getClimateMap() const { return climateMap; }
+class ClimateMapper: commonItems::parser
+{
+  public:
+	ClimateMapper();
+	explicit ClimateMapper(std::istream& theStream);
+
+	[[nodiscard]] const auto& getClimateMap() const { return climateMap; }
```
Comment (Idhrendur) [inline EU4toV2/Source/Mappers/Geography/ClimateMapper.h:14]: Likewise, check if returning the map is the best design.

**Review by Idhrendur — COMMENTED** [review]
Everything but countryMappings is reviewed now.

**Review by Idhrendur — APPROVED** [review]

File: EU4toV2/Source/Mappers/FactoryTypes/FactoryType.h (hunk 8 above)
Comment (Zemurin) [reply EU4toV2/Source/Mappers/FactoryTypes/FactoryType.h:1]: Look at Mappers_FactoryTypeTests, these cover this file and explain the entire stack.

Comment (Zemurin) [reply EU4toV2/Source/Mappers/FactoryTypes/FactoryType.h:1]: btw. On related note, I noticed VS having issues with "show all references", not listing any when I *clearly* have function calls, or listing them partially - those from within the project but not those from tests.

then I open up the tests and suddenly "show all references" works.

Comment (Zemurin) [reply EU4toV2/Source/Mappers/FactoryTypes/FactoryType.h:1]: https://developercommunity.visualstudio.com/content/problem/514082/vs-2019-find-all-references.html

Comment (Zemurin) [reply EU4toV2/Source/Mappers/FactoryTypes/FactoryType.h:1]: This one is 10 days old:
https://developercommunity.visualstudio.com/content/problem/1256537/find-all-referencesgo-to-implementation-returns-no.html

"Closed, not a bug". WTF

**Pull request merged**